### PR TITLE
Neutron mtu master

### PR DESCRIPTION
**Pull request by nicolasbock**
Forward port of https://github.com/crowbar/crowbar-core/pull/503.

### CODE REVIEW

File: chef/data_bags/crowbar/template-network.json [hunk 1]
```diff
@@ -266,7 +267,7 @@
     "network": {
       "crowbar-revision": 0,
       "crowbar-applied": false,
-      "schema-revision": 20,
+      "schema-revision": 21,
```
Comment (toabctl) [inline chef/data_bags/crowbar/template-network.json]: we agreed to start with 100 for Cloud 7

**Comment by toabctl** [comment]
And iirc we usually commit first to master and backport then.

File: chef/data_bags/crowbar/template-network.json (hunk 1 above)
Comment (nicolasbock) [inline chef/data_bags/crowbar/template-network.json]: Ok, I will update that.

**Comment by nicolasbock** [comment]
Ok, will remember that.

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 2]
```diff
@@ -309,6 +312,10 @@ def cinder_volume
           @node[:block_device][@device][:vendor] == "cinder" && @node[:block_device][@device][:model] =~ /^volume-/
         end
 
+        def mtu
+          @node[:mtu] || 1500
```
Comment (vuntz) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: That sounds weird. Why not `@mtu || 1500`?

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 3]
```diff
@@ -224,10 +224,13 @@ def self.lookup_interface_info(node, conduit, intf_to_if_map = nil)
...
 
       class Network
-        attr_reader :name, :address, :broadcast, :mac, :netmask, :subnet, :router, :usage, :vlan, :use_vlan, :interface, :interface_list, :add_bridge, :conduit
+        attr_reader :name, :address, :mtu, :broadcast, :mac, :netmask,
+          :subnet, :router, :usage, :vlan, :use_vlan, :interface,
+          :interface_list, :add_bridge, :conduit
         def initialize(net, data, rintf, interface_list)
           @name = net
           @address = data["address"]
+          @mtu = data["mtu"]
```
Comment (vuntz) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: Why not directly `data["mtu"] || 1500` here? (btw, I'd do `(data["mtu"] || 1500).to_i)` ;-))

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb (hunk 2 above)
Comment (nicolasbock) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: Is the `||` necessary? In `initialize` we test for a non-existing value already.

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 4]
```diff
@@ -309,6 +312,10 @@ def cinder_volume
           @node[:block_device][@device][:vendor] == "cinder" && @node[:block_device][@device][:model] =~ /^volume-/
         end
 
+        def mtu
+          @mtu || 1500
+        end
```
Comment (vuntz) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: We don't need this method now, do we? With `attr_reader`, calling `mtu` will just fetch `@mtu`.

**Comment by vuntz** [comment]
+1

**Comment by toabctl** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Pull request merged**